lightningd/lightningd.c

```diff
@@ -316,92 +503,94 @@ int main(int argc, char *argv[])
 	u32 min_blockheight, max_blockheight;
 	int connectd_gossipd_fd;
 
+	/*~ What happens in strange locales should stay there. */
 	setup_locale();
+	/*~ Every daemon calls this in some form: the hooks are for dumping
+	 * backtraces when we crash (if supported on this platform). */
 	daemon_setup(argv[0], log_backtrace_print, log_backtrace_exit);
+
+	/*~ There's always a battle between what a constructor like this
+	 * should do, and what should be added later by the caller.  In
+	 * general, because we use valgrind heavily for testing, we prefer not
+	 * to intialize unused fields which we expect the caller to set:
+	 * valgrind will warn us if we make decisions based on uninitialized
+	 * variables. */
 	ld = new_lightningd(NULL);
 
 	/* Figure out where our daemons are first. */
 	ld->daemon_dir = find_daemon_dir(ld, argv[0]);
 	if (!ld->daemon_dir)
 		errx(1, "Could not find daemons");
 
+	/*~ The ccan/opt code requires registration then parsing; we
+	 *  mimic this API here, even though they're on separate lines.*/
 	register_opts(ld);
 
-	/* Handle options and config; move to .lightningd */
+	/*~ Handle options and config; move to .lightningd (--lightning-dir) */
 	handle_opts(ld, argc, argv);
 
-	/* Ignore SIGPIPE: we look at our write return values*/
-	signal(SIGPIPE, SIG_IGN);
-
-	/* Make sure we can reach other daemons, and versions match. */
-	test_daemons(ld);
+	/*~ Make sure we can reach the subdaemons, and versions match. */
+	test_subdaemons(ld);
 
-	/* Initialize wallet, now that we are in the correct directory */
+	/*~ Our "wallet" code really wraps the db, which is more than a simple
+	 * bitcoin wallet (though it's that too). */
 	ld->wallet = wallet_new(ld, ld->log, &ld->timers);
+
+	/*~ We keep a filter of scriptpubkeys we're interested in. */
 	ld->owned_txfilter = txfilter_new(ld);
-	ld->topology->wallet = ld->wallet;
 
-	/* We do extra checks in io_loop. */
+	/*~ This is the ccan/io central poll override from above. */
 	io_poll_debug = io_poll_override(io_poll_lightningd);
 
-	/* Set up HSM. */
+	/*~ Set up HSM: it knows our node secret key, so tells us who we are. */
```

Comment:
as the hsmd was mentioned to be the one daemon that needs to be responsive we should explain this a little bit more here?